driver: rtc: microchip: Add RTC-G2

TonyHan11 opened the pull request:
Changes included in this PR:
 - update `drivers/rtc/rtc_sam.c` to support sama7d65
 - add `rtc` to the support list in `sama7d65_curiosity.yaml`
 - enable `tests/drivers/rtc/rtc_api` for sama7d65_curiosity
 
Tested with cases under `tests/drivers/rtc` on sama7d65_curiosity board.

TonyHan11 committed:
tests: drivers: rtc: sam: enable tests for sama7d65_curiosity

Enable tests: tests/drivers/rtc/rtc_api.

Signed-off-by: Tony Han <tony.han@microchip.com>

TonyHan11 committed:
boards: microchip: sama7: enable rtc for sama7 boards

Enable the rtc node in sama7d65_curiosity dts file.
Add rtc to section 'supported' in sama7d65_curiosity.yaml and
sama7g54_ek.yaml files.

Signed-off-by: Tony Han <tony.han@microchip.com>

nandojve commented on drivers/rtc/rtc_sam.c:
```diff
@@ -27,6 +27,68 @@
+ */
+#if defined(CONFIG_SOC_SERIES_SAMA7D6)
+#undef RTC_CR_UPDTIM
+#define RTC_CR_UPDTIM RTC_CR_UPDTIM_Msk
```
You need to bring the change to here to see if everything goes well.
Then when this PR is ready we merge in the HAL and you update the hash here.

TonyHan11 committed:
dts: arm: microchip: add rtc binding file and update sama7d6 dtsi node

Add the binding file for Microchip rtc G2 peripheral.
Update the rtc node in sama7d6.dtsi file.

Signed-off-by: Tony Han <tony.han@microchip.com>

TonyHan11 committed:
drivers: rtc: microchip: introduce G2 rtc driver

Add G2 RTC driver for Microchip RTC Peripherals.

Signed-off-by: Tony Han <tony.han@microchip.com>

bjarki-andreasen reviewed (changes requested):
Could you update the PR title to describe what is changing and why? Its hard to follow why sam0 is going away and mchp_g1 seems to be duplicating a lot from the atmel_sam driver

TonyHan11 commented on drivers/rtc/rtc_mchp_g1.c:
```diff
@@ -0,0 +1,691 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT microchip_rtc_g1
```
As the `G1 RTC` driver is already merged to the `main` branch by #99144 for SAME54, the driver in this PR is renamed to `G2 RTC` to avoid the conflicts.

JarmouniA commented on drivers/rtc/Kconfig.mchp:
```diff
@@ -7,3 +7,11 @@ config RTC_MCHP_G1
+	depends on DT_HAS_MICROCHIP_RTC_G2_ENABLED
+	select SYSCON
+	help
+	  Microchip Real-Time Clock (RTC) G2 driver used on SAM SoC series.
```
That's what `drivers/rtc/rtc_sam.c` is for, no? Is this for a different peripheral?

JarmouniA commented on dts/arm/microchip/sam/sama7/sama7d6/sama7d6.dtsi:
```diff
@@ -65,11 +69,18 @@
 
 		rtc: rtc@e001d800 {
-			compatible = "atmel,sam-rtc";
+			compatible = "microchip,rtc-g2";
```
Why!?

TonyHan11 replied:
Here the new driver `drivers/rtc/rtc_mchp_g2.c` is used instead of using `drivers/rtc/rtc_sam.c`.

JarmouniA replied:
Please clarify that in the binding description.

Also, just for your information, a device DT node can have multiple compatibles, from the most specific to the least.
https://docs.zephyrproject.org/latest/build/dts/bindings-syntax.html#compatible

nandojve replied:
@JarmouniA ,

Microchip have many IPs and those could have evolution over the time. A IP that seems to be the same sometimes are not. As far I know, we don't need to explain what are the differences between the IPs in the devicetree. The information which is relevant to all the new bindings is the `module name="rtc" id="6056" version="0"` which define the IP. That information is relevant to the vendor and Microchip provide correct instructions to their customers when they need to customize a board.

sunil-abraham reviewed (dismissed):
Looks good to me.

nandojve reviewed (changes requested):
@JarmouniA , @bjarki-andreasen ,

Is there anything missing?

TonyHan11 commented:
Rebased by solving conflicts in `boards/microchip/sam/sama7d65_curiosity/sama7d65_curiosity.yaml` introduced by bf9007f3819

@nandojve , @bjarki-andreasen , @Holt-Sun , @sunil-abraham ,
Would you like to review and approve it again? Thank you very much.

JarmouniA reviewed (commented):
Still not addressed https://github.com/zephyrproject-rtos/zephyr/pull/99889#discussion_r2807595636
